### DISCUSSION TITLE
Should ViewSet::getViews() return ModelView instead of View?

### DISCUSSION THREAD

**Discussion by chrikoch** [structurizr/java#208] [Q&A] [answered]
### Question

I wonder why getViews() in https://github.com/structurizr/java/blob/master/structurizr-core/src/com/structurizr/view/ViewSet.java#L527 returns a collection of "View" instead of "ModelView". 

I have a kotlin script run via structrizr-cli and want to get all elements included in a view. But getElements() is only defined on ModelView, not on View. So I can't get the elements of a view :-(

As far as I understand the code, all Elements returned by getViews() are ModelViews, so I suggest to return Collection<ModelView>

**Comment by simonbrowndotje** [comment, marked as answer]
You could just do the Kotlin equivalent of something like:

```
Set<ModelView> views = getViews().stream().filter(v -> v instanceof ModelView).map(v -> (ModelView)v).collect(Collectors.toSet());
```

**Reply by chrikoch** [reply]
Thanks for this super fast answer!

Obviously, this fixed it for me. Still wondering whether collection<ModelView> would be better, but I have a solution for my problem. Thanks!!!

**Reply by simonbrowndotje** [reply]
`getViews()` gets all views, and some of those are not `ModelView` implementations.

So I'm not sure I see how using `Collection` would be better here, and `Collection<ModelView>` isn't appropriate given that not all view types extend `ModelView`.

**Reply by chrikoch** [reply]
Only "Collection" was a typo, sorry for that. You're probably right, thanks for your support!
